kingpin/actors/aws/elb.py

```diff
@@ -232,3 +149,105 @@ def _execute(self):
         utils.clear_repeating_log(repeating_log)
 
         raise gen.Return()
+
+
+class RegisterInstance(base.AWSBaseActor):
+
+    """Add an EC2 instance to a load balancer.
+
+    http://boto.readthedocs.org/en/latest/ref/elb.html
+    #boto.ec2.elb.ELBConnection.register_instances
+    """
+
+    all_options = {
+        'elb': (str, REQUIRED, 'Name of the ELB'),
+        'region': (str, REQUIRED, 'AWS region name, like us-west-2'),
+        'instances': ((str, list), None, (
+            'Instance id, or list of ids. If no value is specified then '
+            'the instance id of the executing machine is used.'))
+    }
+
+    @concurrent.run_on_executor
+    @utils.exception_logger
+    @retry
+    def _add(self, elb, instances):
+        """Invoke elb.register_instances
+
+        This boto function is idempotent, so any retry is OK.
+
+        Args:
+            elb: boto Loadbalancer object
+            instances: list of instance ids.
+        """
+        elb.register_instances(instances)
+
+    @gen.coroutine
+    def _execute(self):
+        elb = yield self._find_elb(self.option('elb'))
+        instances = self.option('instances')
+
+        if not instances:
+            self.log.debug('No instance provided. Using current instance id.')
+            iid = yield self._get_meta_data('instance-id')
```

Comment:
What happens when this fails? Either this method, or the `_get_meta_data()` method should raise a clear exception like 'NotAnEC2Instance', or 'MetadataServerUnavailable' exception. Something pretty clear.